src/neptune_scale/core/components/aggregating_queue.py

```diff
@@ -68,47 +68,33 @@ def commit(self) -> None:
 
     def get(self) -> BatchedOperations:
         start = time.process_time()
-        elements_in_batch: int = 0
-        batch: Optional[RunOperation] = None
+
+        batch_operations: list[(float, RunOperation)] = []
         batch_sequence_id: Optional[int] = None
         batch_timestamp: Optional[float] = None
-        batch_key: Optional[float] = None
+
         batch_bytes: int = 0
+        elements_in_batch: int = 0
         wait_remaining = self._wait_time
 
         # Pull operations off the queue until we either reach the maximum size, or
         # the specified wait time has passed. This way we maximize the potential batch size.
-        while wait_remaining > 0:
+        while True:
             t0 = time.monotonic()
 
             if elements_in_batch >= self._max_elements_in_batch:
+                logger.debug("Batch closed due to limit of elements in batch %s", elements_in_batch)
                 break
 
             element = self._get_next(wait_remaining)
             if element is None:
                 break
 
-            elements_in_batch += 1
-
-            if batch is None:
-                batch = RunOperation()
-                batch.ParseFromString(element.operation)
-
-                batch_sequence_id = element.sequence_id
-                batch_timestamp = element.timestamp
-                batch_key = element.operation_key
-                batch_bytes = len(element.operation)
-
-                self.commit()
-
-                if not element.is_batchable:
-                    logger.debug("Batch closed due to first operation not being batchable")
-                    break
+            if not batch_operations or element.operation_key != batch_operations[-1][0]:
```

Comment:
What would you say to renaming `operation_key` to `batch_key` or similar? I think that would better convey the purpose of this field